WordPress/Classes/Networking/AccountSettingsRemote.swift

```diff
@@ -1,12 +1,78 @@
+import AFNetworking
 import Foundation
+import RxSwift
 
 class AccountSettingsRemote: ServiceRemoteREST {
-    func getSettings(success success: AccountSettings -> Void, failure: ErrorType -> Void) {
+    static let remotes = NSMapTable(keyOptions: .StrongMemory, valueOptions: .WeakMemory)
+
+    /// Returns an AccountSettingsRemote with the given api, reusing a previous
+    /// remote if it exists.
+    static func remoteWithApi(api: WordPressComApi) -> AccountSettingsRemote {
+        // We're hashing on the authToken because we don't want duplicate api
+        // objects for the same account.
+        //
+        // In theory this would be taken care of by the fact that the api comes
+        // from a WPAccount, and since WPAccount is a managed object Core Data
+        // guarantees there's only one of it.
+        // 
+        // However it might be possible that the account gets deallocated and
+        // when it's fetched again it would create a different api object.
+        let key = api.authToken.hashValue
+        // FIXME: not thread safe
+        // @koke 2016-01-21
```

Comment:
I can encapsulate the hashing inside the api object, IIRC this was mostly a convenience since `hashValue` is part of `Hashable`, which is a Swift thing, and `WordPressComApi` is Obj-C. So I think I did this instead of having to create a `WordPressComApi` swift extension just to implement `Hashable`. And also, because `Hashable` is `Equatable` which brings me to this:

> in theory, two instances of WordPressComApi that are logically equivalent (same auth token, same username, etc) are interchangeable.

That would be nice, but `WordPressComApi` is a subclass of `AFHTTPRequestOperationManager`, so it's not just the auth token and username, but its own operation queue with in-progress operations, and all the other inherited AFNetworking state.

Which, by the way, was the reason why we wanted to reuse the same API object per account: so it's all the same operation queue, and we don't send too many concurrent requests to the same host.